cli_tools/common/utils/daisy/daisy_utils.go

```diff
@@ -84,28 +85,44 @@ func UpdateAllInstanceNoExternalIP(workflow *daisy.Workflow, noExternalIP bool)
 	if !noExternalIP {
 		return
 	}
-	updateAllInstanceAccessConfig(workflow, func() []*compute.AccessConfig {
-		return []*compute.AccessConfig{}
-	})
-}
-
-func updateAllInstanceAccessConfig(workflow *daisy.Workflow, accessConfigProvider func() []*compute.AccessConfig) {
-	for _, step := range workflow.Steps {
-		if step.IncludeWorkflow != nil {
-			//recurse into included workflow
-			updateAllInstanceAccessConfig(step.IncludeWorkflow.Workflow, accessConfigProvider)
-		}
+	WorkflowStepIterator(workflow, func(step *daisy.Step) {
 		if step.CreateInstances != nil {
 			for _, instance := range *step.CreateInstances {
 				if instance.Instance.NetworkInterfaces == nil {
 					return
 				}
 				for _, networkInterface := range instance.Instance.NetworkInterfaces {
-					networkInterface.AccessConfigs = accessConfigProvider()
+					networkInterface.AccessConfigs = []*compute.AccessConfig{}
 				}
 			}
 		}
-	}
+	})
+}
+
+// UpdateToUEFICompatible updates workflow resources (disks and images) to be UEFI compatible
+func UpdateToUEFICompatible(workflow *daisy.Workflow) {
+	WorkflowStepIterator(workflow, func(step *daisy.Step) {
+		if step.CreateDisks != nil {
+			for _, disk := range *step.CreateDisks {
+				// for the time being, don't run Debian 9 worker in UEFI mode
+				if strings.Contains(disk.SourceImage, "debian-9-worker") {
```

Comment:
This is hacky - relying on a specific image name is fragile. I feel we should have a tag for CreateInstance to prevent creating worker instance with UEFI.
Going further, why is UEFI not working for debian-9-worker if most of our images will be upgraded to UEFI?

Reply:
This is not image name but image family. As Debian 9 worker gets updated, the family name stays the same.

Alternatively, we could make any disks associated with CreateInstance in import workflows not marked as UEFI, only stand-alone ones (basically customer provided)

I'm not sure if debian-9 worker will work with UEFI. Most likely it will, but we have not done the testing and I would leave it booting using BIOS as it doesn't affect customer GPT formatted images. We can remove this once Debian 9 worker is tested with UEFI.

Reply:
Sounds good, let's remove it later once it's ready due that UEFI is becoming a standard